grinder/core.py

```diff
@@ -1190,13 +1190,23 @@ def __process_current_product_queries(self, product_info: dict) -> None:
             )
 
 
-            masscan_raw_results = self.masscan_scan(
-                hosts=hosts,
-                ports=ports,
-                top_ports=top_ports,
-                rate=rate,
-            )
-            self.__parse_masscan_results(masscan_raw_results, hosts, product_info)
+            try:
+                masscan_raw_results = self.masscan_scan(
+                    hosts=hosts,
+                    ports=ports,
+                    top_ports=top_ports,
+                    rate=rate,
+                )
+                self.__parse_masscan_results(masscan_raw_results, hosts, product_info)
+            except Exception as masscan_exception:
+                if "FAIL: failed to detect MAC address of interface" in str(masscan_exception):
```

Comment:
Why we catch generic `Exception` and look for a substring in the message? Masscan doesn't have its own exception handlers?

Reply:
Fixed